source/postprocess/heating_statistics.cc

```diff
@@ -114,6 +114,11 @@ namespace aspect
 
             this->get_material_model().evaluate(in, out);
 
+            if (this->get_parameters().formulation_temperature_equation
```

Comment:
can you add a one-line comment that says that you're explicitly overwriting the density from the material model here if the formulation requires this?